pkg/target/manager.go

```diff
@@ -338,56 +346,91 @@ func (m *managerImpl) patchTargetWithTarget(target Target) error {
 	})
 }
 
-func (m *managerImpl) SeedClient(ctx context.Context, t Target) (client.Client, error) {
-	t, err := m.getTarget(t)
-	if err != nil {
-		return nil, err
-	}
+func (m *managerImpl) Kubeconfig(ctx context.Context, t Target) ([]byte, error) {
+	if t.ShootName() != "" {
+		return m.getKubeconfig(t, func(client gardenclient.Client) ([]byte, error) {
+			var namespace string
+
+			if t.ProjectName() != "" {
+				project, err := client.GetProject(ctx, t.ProjectName())
+				if err != nil {
+					return nil, err
+				}
+
+				namespace = *project.Spec.Namespace
+			} else {
+				shoot, err := client.FindShoot(ctx, t.AsListOption())
+				if err != nil {
+					return nil, err
+				}
+
+				namespace = shoot.Namespace
+			}
 
-	if t.GardenName() == "" {
-		return nil, ErrNoGardenTargeted
+			return client.GetShootKubeconfig(ctx, namespace, t.ShootName())
+		})
 	}
 
-	if t.SeedName() == "" {
-		return nil, ErrNoSeedTargeted
+	if t.SeedName() != "" {
+		return m.getKubeconfig(t, func(client gardenclient.Client) ([]byte, error) {
+			return client.GetSeedKubeconfig(ctx, t.SeedName())
+		})
 	}
 
-	kubeconfig, err := m.ensureSeedKubeconfig(ctx, t)
-	if err != nil {
-		return nil, err
+	if t.GardenName() != "" {
+		return m.Configuration().Kubeconfig(t.GardenName())
 	}
 
-	return m.clientProvider.FromBytes(kubeconfig)
+	return nil, ErrNoGardenTargeted
 }
 
-func (m *managerImpl) ensureSeedKubeconfig(ctx context.Context, t Target) ([]byte, error) {
-	if kubeconfig, err := m.kubeconfigCache.Read(t); err == nil {
-		return kubeconfig, nil
+func (m *managerImpl) WriteKubeconfig(data []byte) (string, error) {
+	tempDir := filepath.Join(os.TempDir(), "garden")
+
+	fileInfo, err := os.Stat(tempDir)
+	if err != nil || !fileInfo.IsDir() {
+		err := os.Mkdir(tempDir, 0700)
+		if err != nil {
+			return "", fmt.Errorf("failed to create temporary gardenctl directory: %w", err)
+		}
 	}
 
-	gardenClient, err := m.GardenClient(t.GardenName())
-	if err != nil {
-		return nil, fmt.Errorf("failed to create garden cluster client: %w", err)
+	filename := filepath.Join(tempDir, fmt.Sprintf("kubeconfig.%x.yaml", sha256.Sum256(data)))
```

Comment:
as discussed, revert to previous logic (no hash in filename)

Reply:
Implemented the solution we have discussed with the `GCTL_SESSION_ID` which defaults to `ITERM_SESSION_ID` which defaults to a generated v4 uuid.

Reply:
✔️